extensions/data-workspace/src/dialogs/utils.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import * as constants from '../common/constants';
+
+import * as os from 'os';
+import * as path from 'path';
+
+const WINDOWS_INVALID_FILE_CHARS = /[\\/:\*\?"<>\|]/g;
+const UNIX_INVALID_FILE_CHARS = /[\\/]/g;
+const isWindows = os.platform() === 'win32';
+const WINDOWS_FORBIDDEN_NAMES = /^(con|prn|aux|clock\$|nul|lpt[0-9]|com[0-9])$/i;
+
+/**
+ * Returns true if the string is a valid filename
+ * Logic is copied from src\vs\base\common\extpath.ts
+ * @param name filename to check
+ */
+export function isValidBasename(name: string | null | undefined): boolean {
```

Comment:
ideally these validation functions don't need to be added in each of these extensions and could be reused from one common place. Can this just get added in the data workspace extension, then expose it in data-workspace.d.ts so the sql projects extension can use it?

Reply:
If this doesn't get moved to dataworkspace and exposed, then I think src/common/pathUtils.ts would be a more appropriate place than within the dialogs folder.

Reply:
Removed the code in projects extension. And moved the code under src/comon/pathUtils.